test/test-spec-tests.js

```diff
@@ -0,0 +1,129 @@
+    });
+
+    assert.equal(mappedPosition.line, action.generatedLine + 1, `generated line didn't match, expected ${action.generatedLine + 1} got ${mappedPosition.line}`);
+    assert.equal(mappedPosition.column, action.generatedColumn, `generated column didn't match, expected ${action.generatedColumn} got ${mappedPosition.column}`);
+
+    mappedPosition = consumer.originalPositionFor({
+      line: action.generatedLine + 1,
+      column: action.generatedColumn,
+    });
+
+    assert.equal(mappedPosition.line, action.originalLine + 1, `original line didn't match, expected ${action.originalLine + 1} got ${mappedPosition.line}`);
+    assert.equal(mappedPosition.column, action.originalColumn, `original column didn't match, expected ${action.originalColumn} got ${mappedPosition.column}`);
+    assert.equal(mappedPosition.source, action.originalSource, `original source didn't match, expected ${action.originalSource} got ${mappedPosition.source}`);
+    if (action.mappedName)
+      assert.equal(mappedPosition.name, action.mappedName, `mapped name didn't match, expected ${action.mappedName} got ${mappedPosition.name}`);
+  });
+}
+
+async function testTransitiveMappingAction(assert, rawSourceMap, action) {
+  return SourceMapConsumer.with(rawSourceMap, null, async (consumer) => {
+    assert.ok(Array.isArray(action.intermediateMaps), "transitive mapping case requires intermediate maps");
+
+    let mappedPosition = consumer.originalPositionFor({
+      line: action.generatedLine + 1,
+      column: action.generatedColumn,
+    });
+
+    for (let intermediateMapPath of action.intermediateMaps) {
+      const intermediateMap = await readJSON(`./source-map-tests/resources/${intermediateMapPath}`);
+      await SourceMapConsumer.with(intermediateMap, null, (consumer) => {
+        mappedPosition = consumer.originalPositionFor({
+          line: mappedPosition.line,
+          column: mappedPosition.column,
+        });
+      });
+    }
+
+    assert.equal(mappedPosition.line, action.originalLine + 1, `original line didn't match, expected ${action.originalLine + 1} got ${mappedPosition.line}`);
+    assert.equal(mappedPosition.column, action.originalColumn, `original column didn't match, expected ${action.originalColumn} got ${mappedPosition.column}`);
+    assert.equal(mappedPosition.source, action.originalSource, `original source didn't match, expected ${action.originalSource} got ${mappedPosition.source}`);
+  });
+}
+
+for (let testCase of sourceMapSpecTests.tests) {
+  if (skippedTests.includes(testCase.name))
+    continue;
+  exports[`test from source map spec tests, name: ${testCase.name}`] =
+    async function (assert) {
+      const json = await readJSON(`./source-map-tests/resources/${testCase.sourceMapFile}`);
+      let sourceMapFailed = false;
+      try {
+        const map = await new SourceMapConsumer(json);
+        map.eachMapping(() => {});
+        map.destroy();
+      } catch (exn) {
+        if (testCase.sourceMapIsValid)
```

Comment:
```suggestion
        if (testCase.sourceMapShouldBeValid)
```

Reply:
This is not related directly to this patch but it just a tiny suggestion while trying the understanding the code.
`sourceMapShouldBeValid` might make it clearer what the expectation is.